src/Microsoft.Security.Utilities.Core/IdentifiableSecrets.cs

```diff
@@ -78,6 +79,33 @@ public static bool IsBase64UrlEncodingChar(this char ch)
         return encoded == checksumText;
     }
 
+    public static ulong ComputeChecksumSeed(string versionedKeyKind, out string hexFormatted)
+    {
+        hexFormatted = null;
+
+        if (versionedKeyKind == null)
+        {
+            throw new ArgumentNullException(nameof(versionedKeyKind));
+        }
+
+        if (versionedKeyKind.Length != 8 ||
```

Comment:
OK, I misunderstood. So, what's the convention? A string of 7 chars + a single digit that allow for 10 discrete versions?  A string of 6 characters and two digits that allows for 100? Should the API simply create your trailing version and as long as the length of it and the string literal does not exceed 8 chars we are good? It's all looking a little complex. :)

And so I propose to just have people call the API. Let me go back and add some XML comments to clarify the approach.

Also, obviously, feel free to push back or comment on what I've said above.

Reply:
I would suggest 

`public static ulong ComputeChecksumSeed(string keyKind, int version)`

Where keyKind between 1 and 6 characters and version is [0,99].
The implementation would add padding to keyKind as necessary.

Reply:
OK! so I can do this. but if users provide a version >99 then this API will throw. :) is that ok? because we think no one is likely to hit this threshold?

Sounds like a reasonable compromise to what I proposed.

Reply:
> because we think no one is likely to hit this threshold

That and the original suggestion was `RWSeed00` which allowed for two digit growth. I have no opinion on how much version growth should be allowed.